klab.hub/src/main/java/org/integratedmodelling/klab/hub/users/controllers/UserProfileController.java

```diff
@@ -103,6 +120,23 @@ public ResponseEntity<?> getUserProfile(@PathVariable String id) {
 		return new ResponseEntity<>(profile,HttpStatus.ACCEPTED);
 	}
 	
+	@GetMapping(value=API.HUB.USER_BASE_ID_NOAUTH, params = API.HUB.PARAMETERS.USER_GET)
+	public ResponseEntity<?> getUserDataByToken(@PathVariable String id, @RequestParam(API.HUB.PARAMETERS.USER_GET) String setEmail) {
+		TokenVerifyEmailClickback token = null;
+		try {
+			token = (TokenVerifyEmailClickback) tokenService.getAndVerifyToken(id, setEmail, TokenType.verifyEmail);
+		} catch (Exception e) {
+			throw new ActivationTokenFailedException("User Verification token failed");
+		}
+		if (token == null) {
+			throw new ActivationTokenFailedException("User Verification token failed");
```

Comment:
Why don't we use a different error message for the missing token and the exception?

Reply:
Is unreacheable code, I'll delete it.